lib/items/uploadSession.js

```diff
@@ -93,17 +96,23 @@ function uploadSession(params) {
               "Content-Length": chunksToUploadSize,
               "Content-Range": 'bytes ' + uploadedBytes + '-' + (uploadedBytes + chunksToUploadSize - 1) + '/' + params.fileSize
             },
-            body: data,
+            body: payload,
             resolveWithFullResponse: true
           }).then(function (_uploadResponse) {
             uploadResponse = _uploadResponse
             if (uploadResponse.statusCode >= 400) {
               return reject(uploadResponse.body);
             }
 
-            // reset for next chunks
+            // update uploaded bytes
             uploadedBytes += chunksToUploadSize;
-            chunksToUpload = [];
+            /* TODO: 
```

Comment:
@dkatavic I don't know if it's possible with a promise or not but if we are able to register an emitter and emit the value here, the caller would be able to listen for the upload progress. It would be a great edition if possible. Something like this: 
```javascript
oneDriveAPI.items.uploadSimple({
  accessToken: accessToken,
  filename: filename,
  readableStream: readableStream
}).on(progress => console.log(progress)).then((item) => {
// console.log(item);
// returns body of https://dev.onedrive.com/items/upload_put.htm#response
})
```

Otherwise we're good to go.

Reply:
We could add another property that will mark that we want to return Stream that tracks the progress of the upload. One problem I see there is if it is synchronous call it may be more complicated to throw errors for `createUploadSession` API or if there is an error with a Stream, so I'm not sure how messy would it get.

for the future implementation, I have envisioned uploading large files as synchronous wrapper around OneDrive's APIs (like your PR) and having another API around upload session that would mimic OneDrive's API, so it would be a separate call for creating upload session and uploading the files, and we would have the ability to pause, resume and track progress of the upload

Do you have the need to listen to the progress of the upload? I am open for suggestions

Reply:
Upload progress is important in my use case. Anyway, if it's not easy to implement we should go ahead with the merge. I'll figure something out. 
Thanks.

Reply:
I would proceed with this merge and implement streamable progress as a separate PR. In that case, API for the method could be

```javascript
oneDriveAPI.items.uploadSession({
  accessToken: accessToken,
  filename: filename,
  fileSize: fileSize,
  readableStream: readableStream,
  returnProgressStream: true // returns stream that emits number of bytes uploaded instead of resolved promise. Useful for monitoring progress of the upload
}).on('data', (bytesUploaded) => console.log(bytesUploaded))
```

would you mind taking ownership over that PR? I will try to help as much as I can

Reply:
I'll try, but no promises. I'm a little short on time nowadays, because I have to complete my final year project for my university. You can go ahead with the current PR. 
Thanks for all the help!

Reply:
I understand I am a little short on time too. Thank you for your contribution :) , it is a great work. When I get free time I will try to start implementation of the part you need and notify you when I have something ready